pallets/settlement/src/lib.rs

```diff
@@ -1258,9 +1264,9 @@ impl<T: Config> Module<T> {
     fn execute_instruction_retryable(id: InstructionId) -> Result<u32, DispatchError> {
         let result = Self::execute_instruction(id);
         if result.is_ok() {
-            Self::prune_instruction(id);
+            Self::prune_instruction(id, true);
         } else if <InstructionDetails<T>>::contains_key(id) {
-            <InstructionDetails<T>>::mutate(id, |d| d.status = InstructionStatus::Failed);
+            <InstructionStatuses<T>>::mutate(id, |d| *d = InstructionStatus::Failed);
```

Comment:
Use `insert` here instead of `mutate`.  The old code needed to use `mutate` since it was modify a larger value to updated the status.  With the status moved into a new storage item, we don't need to use mutate anymore.